oauth/src/main/java/ch/cyberduck/core/oauth/OAuth2AuthorizationService.java

```diff
@@ -345,13 +354,14 @@ else if(value instanceof Number) {
             return this;
         }
 
-        public TokenResponse toTokenResponse() {
-            return new TokenResponse()
+        public IdTokenResponse toTokenResponse() {
+            return new IdTokenResponse()
                     .setTokenType(tokenType)
                     .setScope(scope)
                     .setExpiresInSeconds(expiresInSeconds)
                     .setAccessToken(accessToken)
-                    .setRefreshToken(refreshToken);
+                    .setRefreshToken(refreshToken)
+                    .setIdToken(idToken);
```

Comment:
This causes a `NullPointerException` when no id token is returned.

Reply:
Good point. Is resolved by the latest commit.